oak/server/rust/oak_runtime/src/runtime/tests.rs

```diff
@@ -140,3 +140,68 @@ fn create_node_more_public_label() {
         }),
     );
 }
+
+#[test]
+fn wait_on_channels_immediately_returns_if_any_channel_is_orphaned() {
+    run_node_body(
+        Label::public_trusted(),
+        Box::new(|runtime| {
+            // Create channels with handles
+            let (write_handle, read_handle_1) = runtime.channel_create(&Label::public_trusted());
+            let (_write_handle, read_handle_2) = runtime.channel_create(&Label::public_trusted());
+
+            // Close the write_handle; this should make the channel Orphaned
+            let result = runtime.clone().channel_close(write_handle);
+            assert_eq!(Ok(()), result);
+
+            // Wait on channel
+            let result = runtime
+                .clone()
+                .wait_on_channels(&[read_handle_1, read_handle_2]);
+            assert!(result.is_ok());
+            assert_eq!(
+                vec![ChannelReadStatus::Orphaned, ChannelReadStatus::NotReady],
+                result?
+            );
```

Comment:
(optional) Does it work to just match the entire thing at once?

```rust
assert_eq!(Ok(vec![ChannelReadStatus::Orphaned, ChannelReadStatus::NotReady]), result)
```

also elsewhere